src/main/java/in/ashwanthkumar/gocd/github/provider/gitlab/GitLabProvider.java

```diff
@@ -0,0 +1,155 @@
+import in.ashwanthkumar.gocd.github.settings.general.DefaultGeneralPluginConfigurationView;
+import in.ashwanthkumar.gocd.github.settings.general.GeneralPluginConfigurationView;
+import in.ashwanthkumar.gocd.github.settings.scm.DefaultScmPluginConfigurationView;
+import in.ashwanthkumar.gocd.github.settings.scm.ScmPluginConfigurationView;
+import in.ashwanthkumar.gocd.github.util.URLUtils;
+import in.ashwanthkumar.utils.func.Function;
+import in.ashwanthkumar.utils.lang.StringUtils;
+import org.gitlab4j.api.GitLabApi;
+import org.gitlab4j.api.GitLabApiException;
+import org.gitlab4j.api.models.Author;
+import org.gitlab4j.api.models.MergeRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+
+public class GitLabProvider implements Provider {
+    private static final Logger LOG = LoggerFactory.getLogger(GitLabProvider.class);
+    public static final String REF_SPEC = "+refs/merge-requests/*/head:refs/remotes/origin/merge-requests/*";
+    public static final String REF_PATTERN = "refs/remotes/origin/merge-requests/";
+
+    @Override
+    public GoPluginIdentifier getPluginId() {
+        return new GoPluginIdentifier("gitlab.pr", Arrays.asList("1.0"));
+    }
+
+    @Override
+    public String getName() {
+        return "Gitlab";
+    }
+
+    @Override
+    public void addConfigData(GitConfig gitConfig) {
+        try {
+            Properties props = GitLabUtils.readPropertyFile();
+            if (StringUtil.isEmpty(gitConfig.getUsername())) {
+                gitConfig.setUsername(props.getProperty("login"));
+            } if (StringUtil.isEmpty(gitConfig.getPassword())) {
+                gitConfig.setPassword(props.getProperty("password"));
+            }
+        } catch (IOException e) {
+            // ignore
+        }
+    }
+
+    @Override
+    public boolean isValidURL(String url) {
+        return new URLUtils().isValidURL(url);
+    }
+
+    @Override
+    public void checkConnection(GitConfig gitConfig) {
+        try {
+            Objects.requireNonNull(loginWith(gitConfig)).getProjectApi().getProjects();
+        } catch (Exception e) {
+            throw new RuntimeException(String.format("check connection failed. %s", e.getMessage()), e);
+        }
+    }
+
+    @Override
+    public String getRefSpec() {
+        return REF_SPEC;
+    }
+
+    @Override
+    public String getRefPattern() {
+        return REF_PATTERN;
+    }
+
+    @Override
+    public void populateRevisionData(GitConfig gitConfig, String prId, String prSHA, Map<String, String> data) {
+        data.put("PR_ID", prId);
+
+        PullRequestStatus prStatus = null;
+        boolean isDisabled = System.getProperty("go.plugin.gitlab.pr.populate-details", "Y").equals("N");
+        LOG.debug("Populating PR details is disabled");
```

Comment:
This log should be inside a `if (isDisabled)` block  isn't?

Reply:
Yes will change it 👍